model/source_http.go

```diff
@@ -14,7 +21,48 @@ type SourceCreateRequest struct {
 	AppCreationWorkflow string  `json:"app_creation_workflow"`
 	AvailabilityStatus  string  `json:"availability_status"`
 
-	SourceTypeID *int64 `json:"source_type_id"`
+	SourceTypeID    *int64      `json:"-"`
+	SourceTypeIDRaw interface{} `json:"source_type_id"`
+}
+
+// Validate validates that the required fields of the SourceCreateRequest request hold proper values. In the specific
+// case of the UUID, if an empty or nil one is provided, a new random UUID is generated and appended to the request.
+func (req *SourceCreateRequest) Validate() error {
+	if req.Name == nil || *req.Name == "" {
+		return fmt.Errorf("name cannot be empty")
+	}
+
+	// If no valid UUID is provided in the request, we generate one
+	if req.Uid == nil || *req.Uid == "" {
+		id := uuid.New()
+		stringId := id.String()
+		req.Uid = &stringId
+	}
+
+	var validWorkflowStatuses = []string{AccountAuth, ManualConfig}
```

Comment:
All right. I've left `validWorkflowStatuses` slice on the `source_http.go`, because after doing a quick lookup I haven't found any more usages of `AppCreationWorkflow` on the repository. I'm thinking that maybe I should move the constants of that array to the `source_http.go` file, since I feel it makes more sense to have everything in the same file, instead of having the constants on `source.go` and then the slice in `source_http.go`. What do you think?